app/controllers/items_controller.rb

```diff
@@ -214,7 +214,9 @@ def set_rights
     dro = object_client.find
     access_additions = CocinaAccess.from_form_value(params[:access_form][:rights])
     updated_access = dro.access.new(access_additions.value!)
-    updated = dro.new(access: updated_access)
+    structural_additions = CocinaStructural.from_form_value(params[:access_form][:rights], dro.structural)
+    updated_structural = dro.structural.new(structural_additions.value!)
```

Comment:
Won't this raise an exception if rights is world?

Reply:
I don't know. I was following the example above and leaning on the existing tests.

Reply:
The existing test is only testing the set to dark case, so can you add a "set to stanford only" case?

Reply:
I'll add a test case for sure.

Reply:
@jcoyne also, isn't this a problem for the CocinaAccess stuff you added above? it can also return `None()`.

Reply:
in the CocinaAccess case `None` is an error. The form should always return something sensible.

Reply:
Yes, it's also a problem there. I'll work on that.

Reply:
The `None` case never occurs on this update form. It's only used for registration. See https://github.com/sul-dlss/argo/pull/2224

Reply:
Aha, I see. OK, I've updated the approach to stop using monads. Take another look?